data_steward/validation/ehr_merge.py

```diff
@@ -0,0 +1,281 @@
+"""
+Combine EHR datasets to form full data set
+
+ * Create a mapping table which arbitrarily maps EHR person_id to RDR person_id and assigns a cdr_id
+ * For each CDM table, load the EHR data and append RDR data (ignore RDR person table)
+ * RDR entity IDs (e.g. visit_occurrence_id, measurement_id) start at 1B
+
+## Notes
+Currently the following environment variables must be set:
+ * BIGQUERY_DATASET_ID: BQ dataset where combined result is stored (e.g. test_join_ehr_rdr)
+ * APPLICATION_ID: GCP project ID (e.g. all-of-us-ehr-dev)
+ * GOOGLE_APPLICATION_CREDENTIALS: path to service account key json file (e.g. /path/to/all-of-us-ehr-dev-abc123.json)
+"""
+import argparse
+import json
+import os
+import logging
+
+import bq_utils
+import resources
+import common
+from googleapiclient.errors import HttpError
+from resources import fields_path
+
+BQ_WAIT_TIME = 10
+PERSON_ID_MAPPING_TABLE = 'person_id_mapping_table'
+VISIT_ID_MAPPING_TABLE = 'visit_id_mapping_table'
+
+PERSON_ID_MAPPING_QUERY_SKELETON = '''
+SELECT
+    ROW_NUMBER() OVER() as global_person_id
+    ,hpo
+    ,person_id as mapping_person_id
+FROM
+(%(union_all_blocks)s);
+'''
+
+PERSON_ID_HPO_BLOCK = '''
+(SELECT
+   person_id
+  ,"%(hpo)s" as hpo
+FROM `%(project_id)s.%(dataset_id)s.%(hpo)s_person`)
+'''
+
+VIST_ID_MAPPING_QUERY_SKELETON = '''
+SELECT
+    ROW_NUMBER() OVER() as global_visit_id
+    ,hpo
+    ,visit_occurrence_id as mapping_visit_id
+FROM
+(%(union_all_blocks)s);
+'''
+
+VISIT_ID_HPO_BLOCK = '''
+(SELECT
+  visit_occurrence_id
+  ,"%(hpo)s" as hpo
+FROM `%(project_id)s.%(dataset_id)s.%(hpo)s_visit_occurrence`)
+'''
+PERSON_ID_MAPPING_TABLE_SUBQUERY = '''
+( SELECT global_person_id, hpo, mapping_person_id
+FROM
+`%(project_id)s.%(dataset_id)s.%(person_id_mapping_table)s`
+)
+person_id_map ON t.person_id = person_id_map.mapping_person_id
+AND
+person_id_map.hpo = '%(hpo)s' '''
+
+VISIT_ID_MAPPING_TABLE_SUBQUERY = '''
+( SELECT global_visit_id, hpo, mapping_visit_id
+FROM
+`%(project_id)s.%(dataset_id)s.%(visit_id_mapping_table)s`
+)
+visit_id_map ON t.visit_occurrence_id = visit_id_map.mapping_visit_id
+AND
+visit_id_map.hpo = '%(hpo)s' '''
+
+TABLE_NAMES = ['person', 'visit_occurrence', 'condition_occurrence', 'procedure_occurrence', 'drug_exposure',
+               'device_exposure', 'measurement', 'observation', 'death']
+
+
+def table_exists(project_id, dataset_id, table_id):
```

Comment:
Is there a reason not to use `bq_utils.table_exists`? If not, please do.

Reply:
`bq_utils.table_exists' uses the default dataset_id.

Reply:
We said we can remove this altogether. We can assume all the tables exist.

Reply:
done.